src/tarball.py

```diff
@@ -56,8 +56,20 @@ def _find_build_config(self, node):
 
     def _update_repo(self, config):
         self.log.info(f"Updating repo for {config.name}")
-        kernelci.build.update_repo(config, self._kdir)
+        try:
+            kernelci.build.update_repo(config, self._kdir)
+        except Exception as err:
+            self.log.error(f"Failed to update repo: {err}")
+            # safeguard, make sure it is git repo
+            if not os.path.exists(os.path.join(self._kdir, '.git')):
+                self.log.error(f"{self._kdir} is not a git repo")
+                return
+            # cleanup the repo and return False, so we try again
```

Comment:
This comment says "return False" and the next return is actually `return True`.

FWIW, I personnally prefer that functions return True on success and False on error (0 == success is a C thing and shouldn't work the same with booleans)

Reply:
Also same for exit/return code of shell programs, 0 == success, so i think i follow Jeny suggestion and just fix a comment.

Reply:
I agree with @a-wai. We usually use `return True` for success and `return False` otherwise.

Reply:
> Also same for exit/return code of shell programs, 0 == success

This is different: 0 is an int, not a bool.

The options are:
* either fully adhere to C/unix conventions and then return an int (which I think don't really make sense in python and is probably a bad idea tbh)
* or use booleans and stick to the most expected behaviour of having `True` for success and `False` for errors